packages/fetchai/connections/soef/connection.py

```diff
@@ -266,13 +269,49 @@ def __init__(
         self.logger = logger
         self._unregister_lock: Optional[asyncio.Lock] = None
 
+    @property
+    def unique_page_address(self) -> Optional[str]:
+        """Get unique page address."""
+        if self._unique_page_address is None:
+            # check if we have it in storage
+            self._unique_page_address = self._get_unique_page_address_from_storage()
+        return self._unique_page_address
+
+    @unique_page_address.setter
+    def unique_page_address(self, unique_page_address: Optional[str]) -> None:
+        """Set the unique page address."""
+        self._unique_page_address = unique_page_address
+        self._set_unique_page_address_to_storage(unique_page_address)
+
+    def _get_unique_page_address_from_storage(self) -> Optional[str]:
+        """Get the unique page address from storage."""
+        if self._token_storage_path is None:
+            return None
+        with open(self._token_storage_path, "r") as f:
+            result = f.read().strip()
```

Comment:
(very minor) Using `pathlib.Path` this would become one line:
```suggestion
        result = Path(self._token_storage_path).read_text().strip()
```

Reply:
Makes it inconsistent, will do it in another context though :)